x-pack/filebeat/input/cel: add support for token authentication (#45359)
elastic/beats: docs/reference/filebeat/filebeat-input-cel.md

The CEL environment enables the [optional types](https://pkg.go.dev/github.com/google/cel-go/cel#OptionalTypes) library using the version defined [here](https://pkg.go.dev/github.com/elastic/mito@v1.22.0/lib#OptionalTypesVersion) and the [two-variable comprehensions extensions](https://pkg.go.dev/github.com/google/cel-go/ext#TwoVarComprehensions) library using the version defined [here](https://pkg.go.dev/github.com/elastic/mito@v1.22.0/lib#TwoVarComprehensionVersion).

> Additionally, it supports authentication via Basic Authentication, Digest Authentication or OAuth2, or token authentication.

Reviewer: If this is being added for the first time in 9.1, add an [inline `applies_to`](https://elastic.github.io/docs-builder/syntax/applies/#inline-annotations):

```suggestion
Additionally, it supports authentication via:

* Basic Authentication
* Digest Authentication
* OAuth2
* Token authentication {applies_to}`stack: ga 9.1.0`
```

Author: @colleenmcginnis How is this rendered in the documentation? I understand that this is part of [cummulative docs](https://elastic.github.io/docs-builder/contribute/cumulative-docs/), but if this were applied consistently we would need to annotate this for nearly all the feature that exist in the input. I have significant concerns about the scalability of this, already in the face of the documentation update diff noise that we have when bumping mito versions in this input.

Reviewer: >This is going to be in 8.19 as well. How are multiple versions handled?

>if this were applied consistently we would need to annotate this for nearly all the feature that exist in the input. 

Cumulative documentation in the Beats docs is just for 9.0+. The 8.19 docs will be available in this dropdown when it is released:

<img height="275" alt="Screenshot 2025-07-15 at 3 54 38 PM" src="https://github.com/user-attachments/assets/f3584e20-87d9-44c9-b383-7279c6f1be53" />  <img  height="295" alt="Screenshot 2025-07-15 at 3 54 50 PM" src="https://github.com/user-attachments/assets/6402e98c-c722-4eaf-943e-37c30bfac8f2" />

>I have significant concerns about the scalability of this, already in the face of the documentation update diff noise that we have when bumping mito versions in this input.

I also have concerns. I have met a few times with other members of the Beats team to talk through concerns, we discussed what we can do in the short term, and I'm committed to advocating for ingest teams' needs around docs (including versioning) in the longer term. I can send you a link to the notes from those conversations via Slack for context if you're interested / want to add anything to it.

>How is this rendered in the documentation?

<img width="510" height="215" alt="Screenshot 2025-07-15 at 4 08 17 PM" src="https://github.com/user-attachments/assets/d3d64899-a61b-48ab-b291-728254ab63e1" />

I'll add screenshots for the others below.

  resource.url: http://localhost
```

> 
> ```yaml
> filebeat.inputs:
> - type: cel
>   auth.token:
>     type: Bearer
>     value: supersecret_bearer_token
>   resource.url: http://localhost
> ```
> 
> ```yaml
> filebeat.inputs:
> - type: cel
>   auth.token:
>     type: Token
>     value: supersecret_token
>   resource.url: http://localhost
> ```
> 

Reviewer: Maybe add some intro text that can be tagged with [inline `applies_to`](https://elastic.github.io/docs-builder/syntax/applies/#inline-annotations) to clarify in which versions these are supported (it might be helpful to add similar into text to the other code blocks above, too):

````suggestion

**Token authentication (Bearer)** {applies_to}`stack: ga 9.1.0`

```yaml
filebeat.inputs:
- type: cel
  auth.token:
    type: Bearer
    value: supersecret_bearer_token
  resource.url: http://localhost
```

**Token authentication (Token)** {applies_to}`stack: ga 9.1.0`

```yaml
filebeat.inputs:
- type: cel
  auth.token:
    type: Token
    value: supersecret_token
  resource.url: http://localhost
```

````

> ### `auth.token.enabled` [_auth_token_enabled]

Reviewer: Use a [section-level `applies_to`](https://elastic.github.io/docs-builder/syntax/applies/#section-annotations) to indicate when this feature was added:

````suggestion
### `auth.token.enabled` [_auth_token_enabled]

```{applies_to}
stack: ga 9.1.0
```
````

When set to `false`, disables the token authentication configuration. Default: `true`.

::::{note}
Token authentication settings are disabled if either `enabled` is set to `false` or the `auth.token` section is missing.
::::

> ### `auth.token.type` [_auth_token_type]

Reviewer: ````suggestion
### `auth.token.type` [_auth_token_type]

```{applies_to}
stack: ga 9.1.0
```
````

The type of token to authenticate with, for example "Token" or "Bearer".

> ### `auth.token.value` [_auth_token_value]

Reviewer: ````suggestion
### `auth.token.value` [_auth_token_value]

```{applies_to}
stack: ga 9.1.0
```
````
